garden-backend-service/src/api/routes/gardens.py

```diff
@@ -99,6 +114,9 @@ async def create_or_replace_garden(
         setattr(existing_garden, key, value)
     try:
         await db.commit()
+        background_tasks.add_task(
```

Comment:
yeah this seems like the natural place to add the task compared to the _create_new_garden call above. I think this is also the "seam" where we want to be checking the SYNC_SEARCH_INDEX flag